packages/typir/src/kinds/class-kind.ts

```diff
@@ -4,25 +4,52 @@
  * terms of the MIT License, which is available in the project root.
  ******************************************************************************/
 
+import { assertUnreachable } from 'langium';
+import { InferenceProblem, InferenceRuleNotApplicable } from '../features/inference.js';
+import { SubTypeProblem } from '../features/subtype.js';
 import { TypeEdge } from '../graph/type-edge.js';
 import { Type } from '../graph/type-node.js';
 import { Typir } from '../typir.js';
-import { IndexedTypeConflict, TypeComparisonStrategy, TypirProblem, compareNameTypesMap, compareValueForConflict, createTypeComparisonStrategy } from '../utils/utils-type-comparison.js';
-import { NameTypePair, toArray } from '../utils/utils.js';
+import { IndexedTypeConflict, MapListConverter, TypeComparisonStrategy, TypirProblem, compareNameTypesMap, compareValueForConflict, createTypeComparisonStrategy } from '../utils/utils-type-comparison.js';
+import { NameTypePair, assertKind, assertTrue, toArray } from '../utils/utils.js';
 import { Kind, isKind } from './kind.js';
 
 export interface ClassKindOptions {
-    structuralTyping: boolean,
+    typing: 'Structural' | 'Nominal',
     /** Values < 0 indicate an arbitrary number of super classes. */
     maximumNumberOfSuperClasses: number,
     subtypeFieldChecking: TypeComparisonStrategy,
+    /** Will be used only internally as prefix for the unique identifiers for class type names. */
+    identifierPrefix: string,
 }
 
 export const ClassKindName = 'ClassKind';
 
+export interface ClassTypeDetails<T1 = unknown, T2 = unknown> { // TODO the generics look very bad!
```

Comment:
I would suggest a composite kind with a class kind as special case. Can show today

Reply:
At the moment, these generics are required for the inference rules of classes.

Reply:
I am looking forward to your PR 🙂